src/gen/primitives/mod.rs

```diff
@@ -0,0 +1,171 @@
+#[macro_use]
+mod macros;
+mod boolean;
+mod string;
+mod duration;
+
+use crate::gen::render::{Renderable, TypeHelperRenderer};
+use genco::prelude::*;
+use paste::paste;
+use uniffi_bindgen::backend::Literal;
+use uniffi_bindgen::interface::{Radix, Type};
+
+pub use boolean::BooleanCodeType;
+pub use string::StringCodeType;
+pub use duration::DurationCodeType;
+
+fn render_literal(literal: &Literal) -> String {
+    fn typed_number(type_: &Type, num_str: String) -> String {
+        match type_ {
+            Type::Int8
+            | Type::UInt8
+            | Type::Int16
+            | Type::UInt16
+            | Type::Int32
+            | Type::UInt32
+            | Type::UInt64
+            | Type::Float32
+            | Type::Float64
+            | Type::Duration => num_str,
+            _ => panic!("Unexpected literal: {} is not a number", num_str),
+        }
+    }
+
+    match literal {
+        Literal::Boolean(v) => format!("{}", v),
+        Literal::String(s) => format!("\"{}\"", s),
```

Comment:
Just noticed. I think dart has a preference for `'`:
 ```suggestion
        Literal::String(s) => format!("'{}'", s),
```